x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/types.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  Schema,
+  IndexJob,
+  SchemaConflicts,
+  FieldCoercionErrors,
+} from '../../../shared/schema/types';
+
+export interface SchemaApiResponse {
+  schema: Schema;
+  mostRecentIndexJob: IndexJob;
+  unconfirmedFields: string[];
+  unsearchedUnconfirmedFields: boolean;
+}
+
+export interface MetaEngineSchemaApiResponse {
+  schema: Schema;
+  fields: SchemaConflicts;
+  conflictingFields: SchemaConflicts;
+}
```

Comment:
Again, just highlighting how different the API responses are based on the engine type. Honestly it feels really weird/not super RESTful to me that we're getting such different responses from the _exact_ same API endpoint_ (except for `schema`) 🤔  I almost would rather be hitting a separate `/meta-engines/` dir or adding a `?type=meta` param just to make it clearer that the data is so different... but I don't think that's worth messing around with right now, maybe another day